scripts/map_tokens.py

```diff
@@ -1,22 +1,53 @@
 #!/usr/bin/env python3
 
 import argparse
-import pickle
+
+import pandas as pd
+from datasets import Dataset
 
 from delphi.eval.token_map import token_map
 from delphi.eval.utils import load_validation_dataset
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser(description="")
+
+    parser.add_argument(
+        "dataset_name",
+        type=str,
+        help="Dataset from huggingface to run token_map on",
+    )
+    parser.add_argument(
+        "--username",
+        type=str,
+        help="Hugging Face API username",
+    )
     parser.add_argument(
-        "dataset_name", help="Dataset from huggingface to run token_map on"
+        "--token",
+        type=str,
+        help="Hugging Face API token",
     )
-    parser.add_argument("--output", help="Output file name", default="token_map.pkl")
     args = parser.parse_args()
 
     dataset = load_validation_dataset(args.dataset_name)
 
-    mapping = token_map(dataset)
+    mapping = token_map(
+        dataset
+    )  # outputs the dictionary: dict[int, list[tuple[int, int]]]
 
-    with open(f"data/{args.output}", "wb") as f:
-        pickle.dump(mapping, file=f)
+    df_dataset = pd.DataFrame.from_dict(
+        {"token_id": list(mapping.keys()), "token_positions": list(mapping.values())}
+    )
+    df_dataset.set_index("token_id", inplace=True)  # Set the token_id as the index
+
+    hf_dataset = Dataset.from_pandas(df_dataset)
```

Comment:
I wonder, are there other constructors for Dataset, like from_dict, from_numpy etc.? Maybe we don't need to go through DataFrame each time.

Reply:
+ `complete_sorted_mapping` is the dictionary where the tokens that don't appear in the dataset are given a key (which dont have in the token_map.py file) and a `None` value.
+ QUESTION: shall we change the `map_token.py` file (specifically the initialization of the `mapping`dictso that the dictionary include all the tokens regardless of whether they appear on the dataset? (this would already give a sorted and complete dictionary for tokens from 0 to 4095)
-> I think this would be more memory efficient by allocating the memory from instantiation, but I'm not sure.

Reply:
it makes sense to change the function in eval/token_map.py to return a list instead of a dict